### PR TITLE
Adding terraform setup for Grafana

### PR DESCRIPTION
**Pull request by dinagraves**
This PR:
- Adds a README
- Adds a cloudbuild.yaml for easy building and deploying
- Updates the terraform installer to automatically build and deploy the dashboard

Outstanding questions:
 - Should we choose a default auth for the dashboard?

### CODE REVIEW

**Comment by dinagraves** [comment]
Fixes https://github.com/GoogleCloudPlatform/fourkeys/issues/168

**Comment by dinagraves** [comment]
The dashboard still needs more style tweaking.

**Review by davidstanke — CHANGES_REQUESTED** [review]
Ooooh, I <3 this! Just one request...

When I open the `dashboard_endpoint`, I get a [grafana home page](https://fourkeys-grafana-dashboard-rrwq5aqzsq-uc.a.run.app/?orgId=1), and then I have to hunt around to find the actual dashboard. Is it possible (I bet not) to generate a link directly to the dashboard? If not, can we add some instructions in the README, and maybe even echo a bit of instruction from the script, telling people how to find it?

**Comment by dinagraves** [comment]
> Ooooh, I <3 this! Just one request...
> 
> When I open the `dashboard_endpoint`, I get a [grafana home page](https://fourkeys-grafana-dashboard-rrwq5aqzsq-uc.a.run.app/?orgId=1), and then I have to hunt around to find the actual dashboard. Is it possible (I bet not) to generate a link directly to the dashboard? If not, can we add some instructions in the README, and maybe even echo a bit of instruction from the script, telling people how to find it?

I think it's possible but I'd have to play with it a bit.  I think you can hardcode a dashboard ID and then link to it that way.

**Comment by dinagraves** [comment]
Updated the URL but didn't test the whole terraform setup again.

File: setup/dashboard/main.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      }
+      service_account_name = var.fourkeys_service_account_email
+    }
+  }
+
+  traffic {
+    percent         = 100
+    latest_revision = true
+  }
+
```
Comment (davidstanke) [inline setup/dashboard/main.tf:25]: Per #216, let's add labels to the resources we create, so they can be bulk-deleted when troubleshooting

```suggestion
  metadata {
    labels = {"created_by":"fourkeys"}
  }
```

File: setup/install.sh [hunk 2]
```diff
@@ -87,9 +90,10 @@ if [ $GENERATE_DATA == "yes" ]; then
 fi
 
 echo "••••••••🔑••🔑••🔑••🔑••••••••"
-echo "configuring Data Studio dashboard…"
-DATASTUDIO_URL="https://datastudio.google.com/datasources/create?connectorId=AKfycbxCOPCqhVOJQlRpOPgJ47dPZNdDu44MXbjsgKw_2-s"
-echo -e "Please visit ${GREEN}$DATASTUDIO_URL${NOCOLOR} to connect your data to the dashboard template."
+echo "configuring Grafana dashboard…"
+DASHBOARD_URL="$(terraform output -raw dashboard_endpoint)d/yVtwoQ4nk/four-keys?orgId=1"
```
Comment (davidstanke) [inline setup/install.sh:94]:
```suggestion
DASHBOARD_URL="$(terraform output -raw dashboard_endpoint)/d/yVtwoQ4nk/four-keys?orgId=1"
```

**Review by davidstanke — CHANGES_REQUESTED** [review]
It works! Just one minor text edit, and a request to add a label to the cloud-run service... TAL at the suggestions.

**Review by davidstanke — APPROVED** [review]

**Pull request merged**